### ISSUE TITLE
Add introspection table with replica frontiers

### ISSUE THREAD

**Issue by umanwizard** [MaterializeInc/materialize#15996]
# Per-Replica Write Frontiers                                                                                                                                                                                           
                                                                                                           
Current introspection sources expose only the computed state and not what the controller/environmentd knows. But this information is used for timestamp selection in the controller. Thus we add an overview table   that contains the compute controllers knowledge about how each replica has progressed for each collection.
                                                                                                           
  We add a builtin table that exposes the *controller* state:
  ```
  materialize=> select * from mz_internal.mz_cluster_replica_frontiers;                                 
  replica_id | export_id     | time                                                           
  -----------+---------------+----------------                                                             
  1          | u13           | 1669730100497                                                               
  2          | u13           | 1669730100000         
...                                                      
  ```                                                                                                      
                                                                                                           
Which follows the naming convention of the related introspection table that exposes the *compute* state
                                                                                                           
  ```                                                                                               
  materialize=> select * from mz_internal.mz_compute_frontiers;                                            
   export_id |     time                                                                            
  -----------+---------------                                                                              
   u13       | 1669730100497                                                                               
   s401      | 1669730101000                                                                               
   s402      | 1669730101000                                                                               
  ....                                                                                                     
  ```                                                                                                                                                                                                                 
                                                                                                           
## Implementation

The implementation follows roughly how `ComputeReplicaMetrics` and `ComputeReplicaHeartbeat` are propagated from the controller to the builtin table. With the difference that we create a dedicated ticker task for initiating the update each second.

Data source: The compute controller maintains a `CollectionState` for each collection it tracks. `CollectionState` contains the write frontier of each replica in `replica_write_frontiers`. This contains all the data we want to publish in the new builtin table.

These write frontiers change frequently, hence we rate limit the update to once a second. We spawn a new tokio task `StatsTicker` in the compute controller. It will set the compute controller to ready each second. The `process` function will return a newly introduced ComputeResponse variant `ComputeReplicaWriteFrontiers(Vec<(ReplicaId, CollectionId, T)>)` . When the coordinator receives this response, it will calculate the builtin table update in the coordinator analogous to how we update [ComputeReplicaMetrics](https://github.com/lluki/materialize/blob/ef34fbea4a4a3ece60934354be58d9ab48909ac5/src/adapter/src/coord/message_handler.rs#L206) .                                               

Matching the behavior of `mz_compute_frontiers`, when encountering an empty write frontier, the result will not be included in the table.

## Out of scope
- It's the 3rd case where we have a ComputeResponse and we essentialy pack this as a table update, thus there is some very similar code in `coord/message_handler.rs`. We should refactor this, but not in this PR.
                                                                                                           
## Open Questions                                                                                                                                                                                                   
- Not clear which rate throttling is a good value, once per second sounds reasonable but that’s based purely on gut feeling.
- Since this table will represent the controller state, it won't contain data immediately. For example when we add a replica, only after it has caught up the new replica will appear in the table. Is this a problem?
- Naming -> discussion [here]( https://materializeinc.slack.com/archives/C02PPB50ZHS/p1670937659099759)

**Comment by lluki** [comment]
I just read this [comment](https://github.com/MaterializeInc/materialize/issues/15921#issuecomment-1309837233) of #15291  which was written one day after this issue. If we have that, we could simply target `mz_internal.mz_compute_frontiers;` in a per replica fashion and be done. Or do you think this will be to not possible in the SQL exporter @umanwizard  ?

**Comment by lluki** [comment]
I just talked with Moritz: The existing introspection exposes the view of the computeds which might be a bit ahead of what environmentd knows. And also it'll only work if the replica is responsive. So it's still a good idea to have it additionally
